be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/MessageHandler.scala

```diff
@@ -20,14 +20,17 @@ trait MessageHandler extends AskPatternConstants {
     */
   def dbActor: AskableActorRef = DbActor.getInstance
 
-  def checkParameters(rpcRequest: JsonRpcRequest, errorMsg: String): Future[GraphMessage] = {
+  def extractParameters[T](rpcRequest: JsonRpcRequest, errorMsg: String): Future[(GraphMessage, Message, Option[T])] = {
     rpcRequest.getParamsMessage match {
-      case Some(_) => Future(Left(rpcRequest))
-      case _       => Future(Right(PipelineError(ErrorCodes.SERVER_ERROR.id, errorMsg, rpcRequest.id)))
+      case Some(_) => 
+        val message: Message = rpcRequest.getParamsMessage.get
+        val data: T = message.decodedData.get.asInstanceOf[T]
+        Future((Left(rpcRequest), message, Some(data)))
+      case _       => Future((Right(PipelineError(ErrorCodes.SERVER_ERROR.id, errorMsg, rpcRequest.id)), null, None))
     }
   }
 
-  def checkLaoChannel(rpcRequest: JsonRpcRequest, errorMsg: String): Future[GraphMessage] = {
+  def extractLaoChannel(rpcRequest: JsonRpcRequest, errorMsg: String): Future[GraphMessage] = {
```

Comment:
Thank you for the change of name! Also what I was thinking was to also do something similiar as `extractParameters` by extracting the laoChannel and returning it in case there is no error. (sorry it was not clear in the previous comments)
So the signature of the function would be something like this (ofc it is up to you what is better for the implementation): 
`def extractLaoChannel(rpc: JsonRpcRequest, errorMsg: String): Future[(GraphMessage, Option[Hash])]`